build_win_lnx.py

```diff
@@ -0,0 +1,176 @@
+            if compiler == "msvc":
+                cts_cmd = cts_solution_cmd_win_x64_msvc_2019
+            elif compiler == "icc":
+                cts_cmd = cts_solution_cmd_win_x64_icc_2019
+        elif bitness == "x86" and msvc_version == "2017":
+            icd_cmd = icd_solution_cmd_win_x86_msvc_2017
+            if compiler == "msvc":
+                cts_cmd = cts_solution_cmd_win_x86_msvc_2017
+            elif compiler == "icc":
+                cts_cmd = cts_solution_cmd_win_x86_icc_2017
+        elif bitness == "x86" and msvc_version == "2019":
+            icd_cmd = icd_solution_cmd_win_x86_msvc_2019
+            if compiler == "msvc":
+                cts_cmd = cts_solution_cmd_win_x86_msvc_2019
+            elif compiler == "icc":
+                cts_cmd = cts_solution_cmd_win_x86_icc_2019
+
+    elif platform[:5] == "linux":
+        print("Linux operating system")
+        compiler = "gcc"
+        msvc_version = "NA"
+        icd_cmd = icd_solution_cmd_linux
+        cts_cmd = cts_solution_cmd_linux
+        build_icd_cmd = 'make -j4'
+        build_cts_cmd = 'make -j4'
+    else:
+        print("Not supported operating system ", platform)
+        return
+
+    if not skip_clone:
+        clone_directories()
+
+    if not skip_icd:
+        print("Build OpenCL-ICD-Loader project")
+        build_project('OpenCL-ICD-Loader',
+                      icd_cmd, build_icd_cmd)
+
+    if not skip_cts:
+        print("Build OpenCL-CTS project")
+        build_project('OpenCL-CTS', cts_cmd, build_cts_cmd)
+
+    print("+++ Summary +++")
+    print("MSVC version:\t ", msvc_version)
+    print("Compiler:\t ", compiler)
+    print("Bitness:\t ", bitness)
+    print("Operating system:\t ", platform)
+    print("ICD solution cmd: \n", icd_cmd)
+    print("ICD build cmd: \n", build_icd_cmd)
+    print("CTS solution cmd: \n", cts_cmd)
+    print("CTS build cmd: \n", build_cts_cmd)
+
+
+def clean_path(path):
+    if os.path.exists(path):
+        print("Cleaning path: ", path)
+        shutil.rmtree(path)
+    else:
+        print("The path: ", path, " does not exist")
+
+
+def process_command_line():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--msvc_version', default="2017", choices=['2017', '2019'],
+                        help='choose msvc version')
+    parser.add_argument('--compiler', default="msvc", choices=['msvc', 'icc'],
+                        help='choose compiler')
+    parser.add_argument('--bitness', default="x64", choices=['x64', 'x86'],
+                        help='choose bitness')
+    parser.add_argument('--skip_clone', action='store_true',  help='skip cloning repositories')
+    parser.add_argument('--skip_icd', action='store_true',  help='skip building ICD project')
+    parser.add_argument('--skip_cts', action='store_true', help='skip building CTS project')
+    parser.add_argument('--build_target', default="ALL_BUILD", help='target to build (only Windows)')
```

Comment:
if this option only works on Windows can this `parser.add_argument` be wrapped in a `platform == 'win32':` so it users don't accidentally expect behaviour from it? Same for the `--msvc_version` and `--compiler` options.